src/es/akaya/src/eu/kanade/tachiyomi/extension/es/akaya/Akaya.kt

```diff
@@ -0,0 +1,194 @@
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Akaya : ParsedHttpSource() {
+
+    override val name: String = "AKAYA"
+
+    override val baseUrl: String = "https://akaya.io"
+
+    override val lang: String = "es"
+
+    override val supportsLatest: Boolean = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .rateLimitHost(baseUrl.toHttpUrl(), 1, 1)
+        .addInterceptor { chain ->
+            val request = chain.request()
+            if (request.url.pathSegments.first() != "serie") return@addInterceptor chain.proceed(request)
+            val response = chain.proceed(request)
+            if (response.request.url.toString().removeSuffix("/") == baseUrl) {
+                throw IOException("Esta serie no se encuentra disponible")
+            }
+            return@addInterceptor response
+        }
+        .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("Referer", "$baseUrl/")
+
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$baseUrl/collection/bd90cb43-9bf2-4759-b8cc-c9e66a526bc6?page=$page", headers)
+
+    override fun popularMangaSelector() = searchMangaSelector()
+
+    override fun popularMangaNextPageSelector() = searchMangaNextPageSelector()
+
+    override fun popularMangaFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun latestUpdatesRequest(page: Int): Request =
+        GET("$baseUrl/collection/0031a504-706c-4666-9782-a4ae30cad973?page=$page", headers)
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        if (query.isEmpty()) return super.fetchSearchManga(page, query, filters)
+        return client.newCall(GET(baseUrl, headers)).asObservableSuccess().map { response ->
+            querySearchMangaParse(response, query)
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = baseUrl.toHttpUrl().newBuilder()
+
+        val order = filters.filterIsInstance<OrderFilter>().first().toUriPart()
+        val genres = filters.filterIsInstance<GenreFilter>().first().state
+            .filter(Genre::state)
+            .map(Genre::id)
+
+        url.addPathSegment(order)
+        if (genres.isNotEmpty()) url.addPathSegment(genres.joinToString(",", "[", "]"))
+
+        url.addQueryParameter("page", page.toString())
+
+        return GET(url.build(), headers)
+    }
+
+    private fun querySearchMangaParse(response: Response, query: String): MangasPage {
+        val document = response.asJsoup()
+        val csrfToken = document.selectFirst("meta[name=csrf-token]")!!.attr("content")
```

Comment:
is it necessary to fetch the token everytime?

Reply:
tbh I have no idea how to edit a requestbody in the interceptor 🫠

Reply:
```
val request = chain.request()
val buffer = Buffer()
request.body?.writeTo(buffer)
    ?: return@addInterceptor chain.proceed(request)

val formData = mutableMapOf<String, String>()
buffer.readString(Charset.defaultCharset()).split("&").forEach {
    val (key, value) = it.split("=")

    formData[key] = value
}
// modify furtther
```

although in this case, you only have the search term, so you can just put it in the url fragment and build the formbody in the interceptor